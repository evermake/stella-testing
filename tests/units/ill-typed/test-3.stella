language core;
extend with #unit-type;

fn Unit::seq(_ : Unit) -> fn(Unit) -> Unit {
  return fn(x : Unit) { return x }
}

fn Bool::seq(_ : Unit) -> fn(Bool) -> Bool {
  return fn(x : Bool) { return if x then false else true }
}

fn main(x : Nat) -> Nat {
	return
    if Bool::seq(Unit::seq(unit)(unit))(Bool::seq(Bool::seq(unit)(false))(Nat::iszero(x)))
    then x
    else succ(x)
}
