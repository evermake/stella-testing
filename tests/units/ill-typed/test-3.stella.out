TYPE ERROR:
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  Unit
but got
  Bool
for expression
  Bool::seq (unit) (false)